<!-- Launch file for the rrt-detector and the assigner -->

<launch>
  <arg name="model" default="burger" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="move_forward_only" default="false"/>
  <arg name="configuration_basename" default="turtlebot3_lds_2d.lua"/>

  <!-- define for the turtlebot environment -->
  <arg name="first_tb3"  default="tb3_0"/>
  <arg name="second_tb3" default="tb3_1"/>
  <arg name="third_tb3"  default="tb3_2"/>
  
  <arg name="first_tb3_x_pos" default=" 0.5"/>
  <arg name="first_tb3_y_pos" default=" 1.7"/>
  <arg name="first_tb3_z_pos" default=" 0.0"/>
  <arg name="first_tb3_yaw"   default=" 0.0"/>

  <arg name="second_tb3_x_pos" default=" 0.5"/>
  <arg name="second_tb3_y_pos" default=" 2.5"/>
  <arg name="second_tb3_z_pos" default=" 0.0"/>
  <arg name="second_tb3_yaw"   default=" 0.0"/>

  <arg name="third_tb3_x_pos" default=" 0.5"/>
  <arg name="third_tb3_y_pos" default=" 3.3"/>
  <arg name="third_tb3_z_pos" default=" 0.0"/>
  <arg name="third_tb3_yaw"   default=" 0.0"/>

  <arg name="dimensions"   default="10,10"/>

<!-- launch gazebos from the data -->
<include file="$(find gazebo_ros)/launch/empty_world.launch">
  <arg name="world_name" value="$(find aws_robomaker_bookstore_world)/worlds/bookstore.world"/>
  <arg name="paused" value="false"/>
  <arg name="use_sim_time" value="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" value="false"/>
  <arg name="debug" value="false"/>
  <arg name="verbose" value="false"/>
</include>

<!-- spawn the robots and related information -->
  <group ns = "/$(arg first_tb3)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="100.0" />
      <param name="use_tf_static" type="bool" value="false" />
      <param name="tf_prefix" value="$(arg first_tb3)" />
    </node>
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg first_tb3) -x $(arg first_tb3_x_pos) -y $(arg first_tb3_y_pos) -z $(arg first_tb3_z_pos) -Y $(arg first_tb3_yaw) -param robot_description" />
    <!-- include gp for the first turtle bot -->
    <include file="$(find seal)/launch/includes/seal_gp.launch">
    </include>
    <!-- include move base for the first turtle bot -->
    <include file="$(find seal)/launch/includes/move_base.launch">
      <arg name="model" value="$(arg model)" />
      <arg name="move_forward_only" value="$(arg move_forward_only)"/>
      <arg name="cmd_vel_topic" value="/$(arg first_tb3)/cmd_vel"/>
      <arg name="namespace" value="$(arg first_tb3)"/>
    </include>
    <!-- include graph for the first turtle bot -->
    <include file="$(find seal)/launch/includes/seal_graph.launch">
    </include>
     <!-- include convex_hull for the first turtle bot -->
     <include file="$(find seal)/launch/includes/seal_convex_hull.launch">
    </include>
<<<<<<< HEAD
=======
    <include file="$(find seal)/launch/includes/seal_gp_to_og.launch">
    </include>
>>>>>>> 61b26445c73e62f9105370972fb1a52bc11c369e

  </group>

  <group ns = "/$(arg second_tb3)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="100.0" />
      <param name="use_tf_static" type="bool" value="false" />
      <param name="tf_prefix" value="$(arg second_tb3)" />
    </node>
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg second_tb3) -x $(arg second_tb3_x_pos) -y $(arg second_tb3_y_pos) -z $(arg second_tb3_z_pos) -Y $(arg second_tb3_yaw) -param robot_description" />
    <!-- include gp for the second turtle bot -->
    <include file="$(find seal)/launch/includes/seal_gp.launch">
    </include>
    <!-- include move base for the second turtle bot -->
    <include file="$(find seal)/launch/includes/move_base.launch">
      <arg name="model" value="$(arg model)" />
      <arg name="move_forward_only" value="$(arg move_forward_only)"/>
      <arg name="cmd_vel_topic" value="/$(arg second_tb3)/cmd_vel"/>
      <arg name="namespace" value="$(arg second_tb3)"/>
    </include>
    <!-- include graph for the second turtle bot -->
    <include file="$(find seal)/launch/includes/seal_graph.launch">
    </include>
    <!-- include convex_hull for the first turtle bot -->
     <include file="$(find seal)/launch/includes/seal_convex_hull.launch">
    </include>
<<<<<<< HEAD
=======
    <include file="$(find seal)/launch/includes/seal_gp_to_og.launch">
    </include>
>>>>>>> 61b26445c73e62f9105370972fb1a52bc11c369e

  </group>

  <group ns = "/$(arg third_tb3)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="100.0" />
      <param name="use_tf_static" type="bool" value="false" />
      <param name="tf_prefix" value="$(arg third_tb3)" />
    </node>
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg third_tb3) -x $(arg third_tb3_x_pos) -y $(arg third_tb3_y_pos) -z $(arg third_tb3_z_pos) -Y $(arg third_tb3_yaw) -param robot_description" />
    <!-- include gp for the third turtle bot -->
    <include file="$(find seal)/launch/includes/seal_gp.launch">
    </include>
    <!-- include move base for the third turtle bot -->
    <include file="$(find seal)/launch/includes/move_base.launch">
      <arg name="model" value="$(arg model)" />
      <arg name="move_forward_only" value="$(arg move_forward_only)"/>
      <arg name="cmd_vel_topic" value="/$(arg third_tb3)/cmd_vel"/>
      <arg name="namespace" value="$(arg third_tb3)"/>
    </include>
    <!-- include graph for the third turtle bot -->
    <include file="$(find seal)/launch/includes/seal_graph.launch">
    </include>
    <!-- include convex_hull for the first turtle bot -->
     <include file="$(find seal)/launch/includes/seal_convex_hull.launch">
    </include>
<<<<<<< HEAD
=======
    <include file="$(find seal)/launch/includes/seal_gp_to_og.launch">
    </include>
>>>>>>> 61b26445c73e62f9105370972fb1a52bc11c369e
  </group>



    <!-- open rviz for visualization purpose -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find seal)/rviz/seal_mapping.rviz">
    </node>


</launch> 
